src/@next/TextArea/TextArea.tsx

```diff
@@ -20,82 +19,80 @@ export type TextAreaProps = Omit<
   onChange: (value: string) => void;
   error?: boolean;
   width?: string;
+  forwardedRef?: RefObject<HTMLTextAreaElement>;
 };
 
-export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
-  function TextArea(
-    {
-      value,
-      rows = 3,
-      maxLength,
-      error = false,
-      disabled = false,
-      width = '520px',
-      onChange,
-      ...props
-    }: TextAreaProps,
-    ref
-  ) {
-    const [charCount, setCharCount] = useState<number>(0);
-    const [isFocused, setIsFocused] = useState<boolean>(false);
-    const isError: boolean = error || (!!maxLength && charCount > maxLength);
+const _TextArea = ({
+  value,
+  rows = 3,
+  maxLength,
+  error = false,
+  disabled = false,
+  width = '520px',
+  onChange,
+  forwardedRef,
+  ...props
+}: TextAreaProps) => {
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const charCount = value.length;
 
-    const localRef = useRef<HTMLTextAreaElement>(null);
-    const setRefs = (node: HTMLTextAreaElement) => {
-      localRef.current = node;
-      if (typeof ref === 'function') {
-        (ref as (instance: HTMLTextAreaElement | null) => void)(node);
-      } else if (ref) {
-        (ref as MutableRefObject<HTMLTextAreaElement | null>).current = node;
-      }
-    };
+  const transformedMaxLength: number = maxLength || undefined;
 
-    useEffect(() => {
-      setCharCount(value.length);
-    }, [value]);
+  const hasMaxLengthEnforced = Boolean(
+    transformedMaxLength && transformedMaxLength > 0
+  );
```

Comment:
Nit: `const hasMaxLengthEnforced = transformedMaxLength > 0`

Reply:
should be fixed in the latest commit,
tl;dr the reason for adding this was that typescript will show linter errors as this value could be undefined too (but i dont get that warning on this codebase though)